Add GPU MoE e2e test script for XLML testing

michelle-yooh opened the pull request:
# Description

This PR adds a test script for GPU MoE e2e in XLML


# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

This is the screenshot of the test result using this script
http://shortn/_dNMK6pqBn9

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

RissyRan reviewed (commented):
LGTM! CC @parambole to take a look at GPU stable stack file

michelle-yooh commented:
For more backgrounds on splitting the maxtext_jax_stable_stack.Dockerfile to gpu and tpu:

This PR adds `transformer_engine_jax==1.13.0` to `requirements_with_jax_stable_stack.txt`
but CUDA is needed in order to install `transformer_engine_jax==1.13.0`
TPU runner has no CUDA installed and led to a failure in image & tests. Hence, it was needed to split the Dockerfiles to utilize the CUDA installation in GPU runner

@parambole please take a look and let me know if this looks good

michelle-yooh commented:
This is the sample run of the test scripts: http://shortn/_MXKnCFfOE0

RissyRan reviewed (commented):
I did not see GPU folder have specific format. How about update file to `end_to_end/gpu/mixtral/test_8x7b.sh` ? For GPU, we don't need to convert ckpt, as it included in TPU scripts. So we probably just need 1 script to run those tests for each model.

RissyRan reviewed (approved):
Just one comment for the file path. Others LGTM! We could add more tests later.

michelle-yooh commented:
> Just one comment for the file path. Others LGTM! We could add more tests later.

@RissyRan I've just addressed your feedback about the file path.
